modular_zubbers/code/modules/security/security_glock/gun.dm

```diff
@@ -1,28 +1,145 @@
+		return ..()
+	magejecting = !magejecting
+	if(!magejecting)
+		balloon_alert(user, "not launching magazines")
+	else
+		balloon_alert(user, "launching empty magazines")
+
+	playsound(user, 'sound/items/weapons/empty.ogg', 100, TRUE)
+	update_appearance()
+	update_item_action_buttons()
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/examine(mob/user)
+	. = ..()
+	. += span_notice("Using it in hand when out of ammo to start a spin reload. This can only be done if you have a spare magazine in or on your belt.")
+	. += span_notice("Spin reloads can be done on the move and need you only to hold your gun for their 2 second duration. ")
+	. += span_notice("If toggled on, spin reloads throw empty magazines in the direction you're facing, dealing damage dependant on their type. ")
+
+// /obj/item/gun/ballistic/automatic/pistol/sec_glock/security //This is what you give to Security Officers.
+// 	name = "\improper C-CK 9x25mm"
+// 	desc = "The Compact Criminal Killer, or C-CK9 for short, is a semi-automatic ballistic pistol meant for regulated station defense. These are normally issued with a special firing pin that only allows firing on code blue or higher."
+// 	icon = 'modular_zubbers/icons/obj/guns/sec_pistol.dmi'
+// 	icon_state = "black"
+// 	pin = /obj/item/firing_pin/alert_level/blue
+// 	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/flathead
+// 	fire_delay = 4
+// 	can_suppress = FALSE
+// 	projectile_damage_multiplier = 0.65
+
+// /obj/item/gun/ballistic/automatic/pistol/sec_glock/security/rubber //This is what you give to cargo packages.
+// 	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/rubber
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/attack_self(mob/living/user)
+	var/obj/item/storage/our_belt = user.get_item_by_slot(ITEM_SLOT_BELT)
+	var/obj/item/ammo_box/magazine/m9mm/security/swapped_mag = locate(/obj/item/ammo_box/magazine/m9mm/security) in our_belt.contents
+	if(!magazine && swapped_mag)
+		if(swapped_mag)
+			if(insert_magazine(user, swapped_mag))
+				SpinAnimation(2, 1)
+				balloon_alert(user, "you spin your gun, loading in a mag!")
+			else
+				balloon_alert(user, "that magazine doesn't fit!")
+		else
+			balloon_alert(user, "no spare magazines in your belt!")
+		return
+	if(!chambered)
+		if(swapped_mag)
+			SpinAnimation(2, 1) //What a badass
+			if(do_after(user, 1 SECONDS, timed_action_flags = ( IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE ), target = src))
+				throw_eject_magazine(user, magejecting)
+		else
+			balloon_alert(user, "no spare magazines in your belt!")
+	if(bolt_type == BOLT_TYPE_NO_BOLT)
+		unload_ammo(user)
+		return
+	drop_bolt_silent(user)
+	if(recent_rack > world.time)
+		return
+	recent_rack = world.time + rack_delay
+	rack_silent(user)
+	return
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/proc/throw_eject_magazine(mob/user, display_message = TRUE, obj/item/ammo_box/magazine/tac_load = null, atom/target)
```

Comment:
You should pass in the magazine you find in the previous proc instead of locating a new one here for deciding what mag to use for reloading